core/database/src/main/java/com/google/samples/apps/nowinandroid/core/database/dao/NewsResourceDao.kt

```diff
@@ -34,29 +34,36 @@ import kotlinx.coroutines.flow.Flow
  */
 @Dao
 interface NewsResourceDao {
-    @Transaction
-    @Query(
-        value = """
-            SELECT * FROM news_resources
-            ORDER BY publish_date DESC
-    """,
-    )
-    fun getNewsResources(): Flow<List<PopulatedNewsResource>>
 
+    /**
+     * Fetches news resources that match the query parameters
+     */
     @Transaction
     @Query(
         value = """
             SELECT * FROM news_resources
-            WHERE id in
-            (
-                SELECT news_resource_id FROM news_resources_topics
-                WHERE topic_id IN (:filterTopicIds)
-            )
+            WHERE 
+                CASE WHEN :useFilterNewsIds
+                    THEN id IN (:filterNewsIds)
+                    ELSE 1
+                END
+             AND
+                CASE WHEN :useFilterTopicIds
+                    THEN id IN
+                        (
+                            SELECT news_resource_id FROM news_resources_topics
+                            WHERE topic_id IN (:filterTopicIds)
+                        )
+                    ELSE 1
+                END
             ORDER BY publish_date DESC
     """,
     )
     fun getNewsResources(
+        useFilterTopicIds: Boolean = false,
```

Comment:
Since the logic to filter on topic and news ids is just whether the supplied list isn't empty we could do this: 
```suggestion
        filterTopicIds: Set<String> = emptySet(),
        useFilterTopicIds: Boolean = filterTopicIds.isNotEmpty(),
        filterNewsIds: Set<String> = emptySet(),
        useFilterNewsIds: Boolean = filterNewsIds.isNotEmpty(),
```

Reply:
I tried this, but it gave a subtle error. If the caller was explicitly wanted to return no news resources with an empty `Set`, they wouldn't be able to.

Ideally we would be able to pass `null` here like we did in the repository, the the generated code will NPE when it tries to bind the SQL args.

Reply:
Ah, what a pain, never mind.